src/vorta/views/repo_add_dialog.py

```diff
@@ -211,6 +214,24 @@ def password_listener(self):
             self.passwordLabel.setText(translate('utils', msg))
             return not bool(msg)
 
+    def cancel_job(self):
+        self.close()
+        try:
+            if self.job:
+                self.job.cancel()
+                logger.debug(f"Cancelled adding repo {self.values['repo_url']}")
+        except Exception as e:
+            logger.error(e)
+
+    # def cancel_queued_job(self):
```

Comment:
Added and commented the method to cancel the job if it is in queue and not picked up